src/Sql/dbo/Stored Procedures/AuthRequest_UpdateMany.sql

```diff
@@ -0,0 +1,30 @@
+CREATE PROCEDURE [dbo].[AuthRequest_UpdateMany]
```

Comment:
❓ : I don't think this accomplishes the same "add or update" methodology of the EF implementation. This method only updates the Auth Requests, it doesn't also add them if they don't exist. Should Auth Requests even be approved if they don't exist?

Reply:
This is a little tricky. I don't *want* this method to be able to create auth requests, for the reasons you laid out and others. So, the Entity Framework implementation is the problem here. *But* I don't know of a way to just do a simple update in EF without the baked in creation logic - do you? 

The EF implementation does include [some handling logic](https://github.com/bitwarden/server/pull/4053/files#diff-7884ebbcbc79ad5dcaa797917eb746ee51e59df458f2424f9ec9094f59519d3fR81-R88) that checks for default IDs on incoming objects and filters them out, which should stop most creations from being possible without a serious logic error somewhere downstream. I also added an assert on https://github.com/bitwarden/server/pull/4053/commits/a8e53d844cd214b35dd440917fb753fe771cf95c to validate that objects without set ids don't make it to the update call.

This won't stop somebody from running `id = new Guid()` and getting created by `UpdateRange`, but I'm not sure how to stop that without doing a costly hit to check all passed in IDs to see if they exist. We can go that route if you want to, or do something else, or leave this as is. Let me know!

Reply:
Entity Framework support is still in Beta so I'm not sure it's worth the trouble to ensure complete accuracy at this moment. 

I did notice that the `dbcontext.UpdateRange(object[])` [does check for default values](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.updaterange?view=efcore-8.0#microsoft-entityframeworkcore-dbcontext-updaterange(system-collections-generic-ienumerable((system-object)))) already in the primary key field and ignore entities that have a `default` value in the PK field. 

I don't think it's worth the effort to address at this time.